pkg/sql/sqlbase/structured.go

```diff
@@ -3091,6 +3156,82 @@ func (desc *MutableTableDescriptor) MakeMutationComplete(m DescriptorMutation) e
 					return err
 				}
 			}
+		case *DescriptorMutation_ComputedColumnSwap:
+			args := t.ComputedColumnSwap
+
+			oldCol, err := desc.FindColumnByID(args.OldColumnId)
+			if err != nil {
+				return err
+			}
+			newCol, err := desc.FindColumnByID(args.NewColumnId)
+			if err != nil {
+				return err
+			}
+
+			newCol.Hidden = false
+			oldCol.Hidden = true
+
+			newCol.ComputeExpr = nil
+			// oldCol should still have new values written to newCol in case
+			// it has a read from previous version.
+			oldCol.ComputeExpr = proto.String(
+				fmt.Sprintf("%s::%s", oldCol.Name, oldCol.Type.String()))
+
+
+			//Rename newCol to oldCol.Name.
+			//Generate unique name for old column.
+			nameExists := func(name string) bool {
+				_, _, err := desc.FindColumnByName(tree.Name(name))
+				return err == nil
+			}
+
+			uniqueName := GenerateUniqueConstraintName(
+				newCol.Name,
+				nameExists,
+			)
+
+			oldColName := oldCol.Name
+			newColName := newCol.Name
+			desc.RenameColumn(oldCol, uniqueName)
+			desc.RenameColumn(newCol, oldColName)
+			desc.RenameColumn(oldCol, newColName)
+
+			// Handle indexes as well.
+
+			// Update Column Families, new Col is not in a family.
+			//for i := range desc.Families {
```

Comment:
I dont think you need any of this -- you just have to make sure to add the newCol to the family of the old column, and put it in the position of the old column.